gossip/src/cluster_info.rs

```diff
@@ -663,6 +663,12 @@ impl ClusterInfo {
         self.my_contact_info.read().unwrap().shred_version
     }
 
+    /// Sets the node's own TPU address in gossip cluster info.
+    pub fn set_tpu_address(&self, tpu_address: SocketAddr) {
+        self.my_contact_info.write().unwrap().tpu = tpu_address;
+        self.push_self(&HashMap::new(), None);
```

Comment:
I'm not sure whether we'd need `self.insert_self` as well. Advice would be appreciated